website/docs/beginner/env.md

```diff
@@ -3,51 +3,128 @@ sidebar_position: 4
 sidebar_label: Setup .env
 ---
 
-# How to Set Up a .env File with TypeScript in Node.js
+# How to Set Up a `.env` File with Node.js, Deno or Bun
 
-## Step 1: Install Required Packages
+Each runtime has it's own way of handling the `.env` file, we will see how to do it with Node.js, Deno and Bun
 
-Before you can set up a `.env` file with TypeScript in Node.js, you need to install the required packages. Run the following command in your project's directory:
+## Setup
 
-`npm install dotenv @types/dotenv --save-dev`
-
-## Step 2: Create a .env File
-
-1. Create a new file named `.env` in the root directory of your project.
-2. Add your environment variables to the file in the following format:
-
-VAR_NAME=value
+Create a `.env` file we will be loading into our runtime environment and add some variables to it in the following format: `VAR_NAME=value`
 
 For example:
 
-```js
+```ini
 DB_HOST = localhost
 DB_USER = admin
 DB_PASS = password123
 ```
 
-## Step 3: Load Environment Variables
+## Node.js
+
+### Loading the `.env` file
+
+Now that we created this file we need to load it into the environment variables of the node process, we have 2 alternative
+
+1. Use the build-in `--env-file` option (requires Node.js v20.6+)
+1. Use the `dotenv` package
+
+#### --env-file option
+
+You need to update the command line you use for running node with the following **node** option and passing it the file name for the `.env` file, the file is resolved based on the directory you are running node from
+
+Here is an example:
+
+```sh
+node --env-file=.env index.js
+```
+
+You need to add this flag before the file you want to run also if you have multiple `.env` file you want to include you can add multiple `--env-file`
 
-1. In your TypeScript file, import the `dotenv` package:
+#### `dotenv` package
+
+If you don't want or can't use the `--env-file` option you can do it with the `dotenv` package
+
+First of all you need to install the `dotenv` package along it's type if you are using TypeScript
+
+`npm install dotenv @types/dotenv --save-dev`
+_or using yarn, pnpm or bun_
+
+We now need to load the `.env` from our code so let's add the following code to your typescript file:
```

Comment:
you can also use dotenv on command line:
```
node -r dotenv/config script.js
```

This will load the `.env` from where-ever you're running it.

Reply:
i know but importing the dotenv package is more known way to do this, i guess it could be mentioned tho

Reply:
Just wondering if it's worth mentioning a non-code change based alternative? This is, funny enough, how I started using dotenv a while ago so I don't have to use `.env` files on prod and only for dev

Reply:
If we want to be sincere: before i started using --env-file from node (what i do prefer now) i was using `import 'dotenv/config'` that is basically both. So you can argue for all 3, but the import with side effects i would avoid in a beginned scoped docs, and using preloading is adding yet another option to dotenv and --env-file that it is not that big of a deal honestly and in dotenv readme itself you need you need to scroll a bit to find it

Reply:
fine by me! just wanted to throw another option out there, but your stance makes sense. I haven't upgraded to 20 yet, can't wait to not have to use dotenv.